coalib/results/Diff.py

```diff
@@ -175,16 +175,31 @@ def affected_code(self, filename):
         """
         return list(diff.range(filename) for diff in self.split_diff())
 
-    def split_diff(self):
+    def split_diff(self, distance=1):
         """
         Splits this diff into small pieces, such that several continuously
         altered lines are still together in one diff. All subdiffs will be
         yielded.
+
+        A diff like this with changes being together closely won't be splitted:
+
+        >>> diff = Diff.from_string_arrays([     'b', 'c', 'e'],
+        ...                                ['a', 'b', 'd', 'f'])
+        >>> len(list(diff.split_diff()))
+        1
+
+        If we set the distance to 0, it will be splitted:
+
+        >>> len(list(diff.split_diff(distance=0)))
+        2
+
+        :param distance: Number of unchanged lines that are allowed in between
+                         two changed lines so they get yielded as one diff.
```

Comment:
shouldn't we assert that distance > 0, i don't think it matters i'm just trying to be logical, we should have a generic test case in case we are allowing that